File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/views/home/fragments/contact_fragment/official_account/OfficialAccountAdapter.java
package vng.zalo.tdtai.zalo.zalo.views.home.fragments.contact_fragment.official_account;

import android.net.Uri;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import vng.zalo.tdtai.zalo.R;
import vng.zalo.tdtai.zalo.zalo.models.DataModel;
import vng.zalo.tdtai.zalo.zalo.models.RoomItem;
import vng.zalo.tdtai.zalo.zalo.utils.ModelViewHolder;

public class OfficialAccountAdapter extends ListAdapter<RoomItem, OfficialAccountAdapter.OfficialAccountViewHolder> {
    private OfficialAccountSubFragment officialAccountSubFragment;

    OfficialAccountAdapter(OfficialAccountSubFragment officialAccountSubFragment, @NonNull DiffUtil.ItemCallback<RoomItem> diffCallback) {
        super(diffCallback);
        this.officialAccountSubFragment = officialAccountSubFragment;
    }

    @NonNull
    @Override
    public OfficialAccountViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_official_account, parent, false);
//        view.setOnClickListener(officialAccountSubFragment);
        return new OfficialAccountViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull OfficialAccountViewHolder holder, int position) {
        holder.bind(position);
    }

    @Override
    public int getItemCount() {
        return getCurrentList().size();
    }

    class OfficialAccountViewHolder extends RecyclerView.ViewHolder implements ModelViewHolder {
        View itemView;
        TextView nameTextView;
        ImageView avatarImgView;
        ImageView iconImgView;

        OfficialAccountViewHolder(@NonNull View itemView) {
            super(itemView);
            this.itemView = itemView;
            nameTextView = itemView.findViewById(R.id.nameTextView);
            avatarImgView = itemView.findViewById(R.id.avatarImgView);
            iconImgView = itemView.findViewById(R.id.iconTextView);
        }

        @Override
        public void bind(int position) {
            itemView.setOnClickListener(officialAccountSubFragment);
            RoomItem room = getItem(position);

            nameTextView.setText(room.name);

            new Picasso.Builder(avatarImgView.getContext()).listener(new Picasso.Listener() {
                @Override
                public void onImageLoadFailed(Picasso picasso, Uri uri, Exception exception) {
                    Log.e(OfficialAccountAdapter.this.getClass().getSimpleName(), exception.getMessage());
                    exception.printStackTrace();
                }
            }).build().load(room.avatar)
                    .fit()
                    .into(avatarImgView);
        }
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/views/home/fragments/contact_fragment/official_account_sub_fragment/OfficialAccountAdapter.kt
package vng.zalo.tdtai.zalo.zalo.views.home.fragments.contact_fragment.official_account_sub_fragment

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.item_official_account.view.*
import vng.zalo.tdtai.zalo.R
import vng.zalo.tdtai.zalo.zalo.models.RoomItem
import vng.zalo.tdtai.zalo.zalo.utils.ModelViewHolder

class OfficialAccountAdapter (private val officialAccountSubFragment: OfficialAccountSubFragment, diffCallback: DiffUtil.ItemCallback<RoomItem>) : ListAdapter<RoomItem, OfficialAccountAdapter.OfficialAccountViewHolder>(diffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OfficialAccountViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_official_account, parent, false)
        return OfficialAccountViewHolder(view)
    }

    override fun onBindViewHolder(holder: OfficialAccountViewHolder, position: Int) {
        holder.bind(position)
    }

    override fun getItemCount(): Int {
        return currentList.size
    }

    inner class OfficialAccountViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), ModelViewHolder {

        override fun bind(position: Int) {
            itemView.setOnClickListener(officialAccountSubFragment)
            val room = getItem(position)

            itemView.nameTextView.text = room.name

            Picasso.Builder(itemView.avatarImgView.context)
                    .build().load(room.avatar)
                    .fit()
                    .into(itemView.avatarImgView)
        }
    }
}